pkg/controller/master/upgrade/upgrade_controller.go

```diff
@@ -166,10 +172,34 @@ func (h *upgradeHandler) OnChanged(key string, upgrade *harvesterv1.Upgrade) (*h
 		toUpdate.Status.SingleNode = singleNode
 
 		repoInfo, err := repo.getInfo()
+		// retry repo.getInfo by requeueing
 		if err != nil {
-			setUpgradeCompletedCondition(toUpdate, StateFailed, corev1.ConditionFalse, err.Error(), "")
-			return h.upgradeClient.Update(toUpdate)
+			retryCount := 0
```

Comment:
maybe we can use the [k8s retry utils](https://pkg.go.dev/k8s.io/client-go/util/retry) instead

Reply:
Updated, please check again. Thanks!